src/data/equiv/basic.lean

```diff
@@ -1289,12 +1289,20 @@ def subtype_subtype_equiv_subtype_exists {α : Type u} (p : α → Prop) (q : su
   λ⟨a, ha⟩, ⟨⟨a, ha.cases_on $ assume h _, h⟩, by { cases ha, exact ha_h }⟩,
   assume ⟨⟨a, ha⟩, h⟩, rfl, assume ⟨a, h₁, h₂⟩, rfl⟩
 
+@[simp] lemma subtype_subtype_equiv_subtype_exists_apply {α : Type u} (p : α → Prop)
+  (q : subtype p → Prop) (a : _) : (subtype_subtype_equiv_subtype_exists p q a).val = a.val.val :=
```

Comment:
I think we normally use coercion instead of `.val` when stating things about subtypes.